bin/oneoff/backfill_data/backfill_csa_section_course_id

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env ruby
+
+require_relative '../../../dashboard/config/environment'
+
+CDO.log = Logger.new(STDOUT)
+ActiveRecord::Base.record_timestamps = false
+
+options = {actually_update: false}
+OptionParser.new do |opts|
+  opts.banner = "Usage: #{File.basename(__FILE__)} [options]"
+  opts.on('-u', '--actually-update', 'Actually perform the update.') do
+    options[:actually_update] = true
+  end
+  opts.on('-h', '--help', 'Add -u to perform the update.') do
+    puts opts
+    exit
+  end
+end.parse!
+CDO.log.info "Called with options: #{options}"
+
+pilot_names = %w(csa-pilot csa-pilot-facilitator)
+updated = []
+
+pilot_names.each do |pilot_name|
+  CDO.log.info "fixing pilot #{pilot_name}."
+  pilot = UnitGroup.find_by_name(pilot_name)
+  script_ids = UnitGroupUnit.where(course_id: pilot.id).pluck(:script_id)
+  sections = Section.where(course_id: nil, script_id: script_ids)
+  CDO.log.info "found #{sections.length} sections to update."
+  sections.each do |section|
+    if options[:actually_update]
+      section.course_id = pilot.id
+      section.save!
```

Comment:
If an individual Section can’t be updated successfully, I think this entire script will exit? Is it worth it to `rescue` inside this loop, log the Section we couldn’t update, and then move on to the next Section? The dataset is small and updating a single column on the model is unlikely to fail for a subset of Sections, so possibly not necessary.

Reply:
added a rescue!